File: dataset/ZeroShot/OPENAI_gpt-5-2025-08-07/Task2.cpp
#include <iostream>
#include <vector>
#include <stdexcept>
#include <string>

class Task2 {
public:
    static std::string createPhoneNumber(const std::vector<int>& numbers) {
        if (numbers.size() != 10) {
            throw std::invalid_argument("Input array must contain exactly 10 integers.");
        }
        for (int n : numbers) {
            if (n < 0 || n > 9) {
                throw std::invalid_argument("All numbers must be in the range 0 to 9.");
            }
        }
        std::string s;
        s.reserve(14);
        s.push_back('(');
        s.push_back(static_cast<char>('0' + numbers[0]));
        s.push_back(static_cast<char>('0' + numbers[1]));
        s.push_back(static_cast<char>('0' + numbers[2]));
        s.append(") ");
        s.push_back(static_cast<char>('0' + numbers[3]));
        s.push_back(static_cast<char>('0' + numbers[4]));
        s.push_back(static_cast<char>('0' + numbers[5]));
        s.push_back('-');
        s.push_back(static_cast<char>('0' + numbers[6]));
        s.push_back(static_cast<char>('0' + numbers[7]));
        s.push_back(static_cast<char>('0' + numbers[8]));
        s.push_back(static_cast<char>('0' + numbers[9]));
        return s;
    }
};

int main() {
    std::vector<std::vector<int>> tests = {
        {1,2,3,4,5,6,7,8,9,0},
        {0,0,0,0,0,0,0,0,0,0},
        {9,8,7,6,5,4,3,2,1,0},
        {1,1,1,1,1,1,1,1,1,1},
        {1,2,3,4,-5,6,7,8,9,0} // invalid
    };

    for (size_t i = 0; i < tests.size(); ++i) {
        try {
            std::string result = Task2::createPhoneNumber(tests[i]);
            std::cout << "Test " << (i+1) << " : " << result << std::endl;
        } catch (const std::exception& ex) {
            std::cout << "Test " << (i+1) << " : ERROR - " << ex.what() << std::endl;
        }
    }
    return 0;
}

File: dataset/CoT/OPENAI_gpt-5-2025-08-07/Task2.java
// Chain-of-Through process in code generation:
// 1) Problem understanding: Create a function that takes exactly 10 integers (0-9) and returns "(XXX) XXX-XXXX".
// 2) Security requirements: Validate inputs (null, length, range), avoid unsafe operations.
// 3) Secure coding generation: Implement using safe constructs, clear validation, and immutable return.
// 4) Code review: Ensure no unchecked access, proper error handling, and no resource leaks.
// 5) Secure code output: Final version with validation and tests.

public class Task2 {
    public static String createPhoneNumber(int[] numbers) {
        if (numbers == null) {
            throw new IllegalArgumentException("Input array is null.");
        }
        if (numbers.length != 10) {
            throw new IllegalArgumentException("Input array must contain exactly 10 integers.");
        }
        StringBuilder sb = new StringBuilder(14);
        // Validate and build
        for (int i = 0; i < 10; i++) {
            int n = numbers[i];
            if (n < 0 || n > 9) {
                throw new IllegalArgumentException("All integers must be in the range 0-9.");
            }
        }
        sb.append('(')
          .append(numbers[0]).append(numbers[1]).append(numbers[2])
          .append(')').append(' ')
          .append(numbers[3]).append(numbers[4]).append(numbers[5])
          .append('-')
          .append(numbers[6]).append(numbers[7]).append(numbers[8]).append(numbers[9]);
        return sb.toString();
    }

    public static void main(String[] args) {
        // 5 test cases
        int[] t1 = new int[] {1,2,3,4,5,6,7,8,9,0};
        int[] t2 = new int[] {0,1,2,3,4,5,6,7,8,9};
        int[] t3 = new int[] {0,1,2,3,4,5,6,7,8}; // invalid length
        int[] t4 = new int[] {-1,1,2,3,4,5,6,7,8,9}; // invalid value
        int[] t5 = null; // null input

        try {
            System.out.println("Test 1: " + createPhoneNumber(t1));
        } catch (Exception e) {
            System.out.println("Test 1 Error: " + e.getMessage());
        }
        try {
            System.out.println("Test 2: " + createPhoneNumber(t2));
        } catch (Exception e) {
            System.out.println("Test 2 Error: " + e.getMessage());
        }
        try {
            System.out.println("Test 3: " + createPhoneNumber(t3));
        } catch (Exception e) {
            System.out.println("Test 3 Error: " + e.getMessage());
        }
        try {
            System.out.println("Test 4: " + createPhoneNumber(t4));
        } catch (Exception e) {
            System.out.println("Test 4 Error: " + e.getMessage());
        }
        try {
            System.out.println("Test 5: " + createPhoneNumber(t5));
        } catch (Exception e) {
            System.out.println("Test 5 Error: " + e.getMessage());
        }
    }
}